### ISSUE TITLE
Delete command from the history?

### ISSUE THREAD

**Issue by Suncatcher** [microsoft/terminal#13123]
# Description of the new feature/enhancement
Sometime we need to clean out unnecessary commands from the history, but keep all the rest, so it shouldn't be a total history wipe. 

# Proposed technical implementation details (optional)

In *nix we have a very nice and nifty command that does exactly that:

    history -d linenumber

We should have the same capability in Windows Terminal as well.

**Comment by zadjii-msft** [comment]
Commandline history is something that is largely driven on a per-shell basis. Which shell are you using? `cmd`? `powershell`/? 

I bet that PowerShell already has a similar feature. `cmd` probably not as much, though there is `GetConsoleHistoryInfo`/`SetConsoleHistoryInfo`, which I'm sure `doskey` can modify in a more user friendly way.

**Comment by Suncatcher** [comment]
> Which shell are you using? cmd? powershell?

PS 7.2.3.

I believe cmd has no history capability, so it is pretty irrelevant for this feature request. Unless of course Windows Terminal adds own history on top of `cmd`

**Comment by Suncatcher** [comment]
> I bet that PowerShell already has a similar feature

Yes it already has. It is `Clear-History -ID 12, 20` cmdlet. But I observe it behaves weird on the Terminal. Example:

![image](https://user-images.githubusercontent.com/6388034/169035300-ee60bccd-bc8d-4f08-9c90-c0dd2bbdc6d9.png)

I use Get-History cmdlet to het the commands IDs. but it fetches me the history only of **current terminal session**, however by arrow keys I can access all commands from the past. How to access all IDs?

**Comment by Suncatcher** [comment]
> And you can empty or delete $env:AppData\Microsoft\Windows\PowerShell\PSReadLine\ConsoleHost_history.txt

thanks, that worked. I'm closing this ticket, but hotkey for this would be much more convenient than editing this file.

**Comment by DHowett** [comment]
> hotkey for this would be much more convenient than editing this file

Agreed! That is a request you should file on https://github.com/PowerShell/PSReadLine, though, as they control the storage and manipulation of history in PowerShell.
